lib/MetaCPAN/Model/Tarball.pm

```diff
@@ -0,0 +1,177 @@
+package MetaCPAN::Model::Tarball;
+
+use Archive::Any;
+use CPAN::DistnameInfo ();
+use DateTime           ();
+use DDP;
+use Log::Contextual qw( :log :dlog );
+use MetaCPAN::Types qw(ArrayRef File Dir);
+use Moose;
+use MooseX::StrictConstructor;
+use Path::Class qw(file dir);
+
+has tarball => (
+    is       => 'ro',
+    isa      => File,
+    required => 1,
+    coerce   => 1,
+);
+
+has files => (
+    is       => 'ro',
+    isa      => ArrayRef,
+    init_arg => undef,
+    lazy     => 1,
+    builder  => '_build_files'
+);
+
+has base_dir => (
+    is      => 'ro',
+    isa     => Dir,
+    coerce  => 1,
+    default => '/tmp',
+);
+
+has date => (
+    is  => 'ro',
+    isa => 'DateTime',
+);
+
+has index => (
+    is         => 'ro',
```

Comment:
I am still figuring which one what is, but again, there are not too many attributes? When I add a new file, I need too much information and I don't know how else to carry that.